x/logic/fs/file.go

```diff
@@ -0,0 +1,63 @@
+package fs
+
+import (
+	"bytes"
+	"io/fs"
+	"net/url"
+	"time"
+)
+
+type VirtualFile struct {
+	reader  *bytes.Reader
+	uri     *url.URL
+	modTime time.Time
+}
+
+var (
+	_ fs.File     = (*VirtualFile)(nil)
+	_ fs.FileInfo = (*VirtualFile)(nil)
+)
+
+func NewObject(src []byte, uri *url.URL, modTime time.Time) VirtualFile {
```

Comment:
```suggestion
func NewVirtualFile(src []byte, uri *url.URL, modTime time.Time) VirtualFile {
```

This one has been missed :)

Reply:
😭😭🤦